explainer-datamodel.md

```diff
@@ -119,66 +131,247 @@ FormattedText.format( [ "The quick ",
+  let col1Lines = lineIter.next(0).value; // process all the lines and return them in an array
+  
+  // Rough-out the line at which to move the rest to column 2
+  let cutLineIndex = Math.floor( ratio * lineIter.lineCount );
+  // Discard the lines (inclusive) after this cut line from column 1's lines.
+  col1Lines.splice( cutLineIndex );
+  
+  // Reset the iterator to the cut-line index, adjust the iterator's inlineSize constraint, and
+  //  get the rest of the lines
+  lineIter.reset( cutLineIndex );
+  lineIter.inlineSize = col2_inlineSize;
+  let col2Lines = lineIter.next(0).value;
+  
+  // Assign positions for each of the lines in both columns and render (left-justified)
+  // First column location
+  let xPos = col1Box.x;
+  let yPos = col1Box.y;
+  for ( let line of col1Lines ) {
+    renderFunc( line, xPos, yPos );
+    yPos += line.height;
+  }
+  // Second column location
+  xPos = col2Box.x;
+  yPos = col2Box.y;
+  for ( let line of col2Lines ) {
+    renderFunc( line, xPos, yPos );
+    yPos += line.height;
+  }
+}
+```
+
 ## Comparison to HTML
 
-`format` is used to drive the web platform's layout engine, but using the JavaScript-based data
-model described above. Therefore, the output of `format` should be the equivalent to what can be
-already performed in HTML using simple elements like `div` and `span`. The following two expressions
-are functionally equivalent, with the exception that the result of `format` has not been rendered
-(and thus can't be visualized yet):
+The formatting functions are used to drive the web platform's layout engine, therefore, the output of
+`format` and `lines` should be the equivalent to what can be already performed in HTML using simple 
+elements like `div` and `span`. The following two expressions are functionally equivalent, with the
+exception that the result of `format` has not been rendered (how to
+[render it is described separately](explainer-rendering.md)):
 
 ```js
 FormattedText.format( [ "The quick ",
                         { text: "brown", style: "color: brown; font-weight: bold" },
                         " fox jumps over the lazy dog"
-                      ], null, { width: 150 } );
-```
-
-```html
-<div style="width:150px">
-  <span id="meta_object">
-    The quick <span style="color: brown; font-weight: bold">brown</span> fox jumps over the lazy dog
-  </span>
-</div>
+                      ], null, 150, 100 );
 ```
 
-Above, the `<div>` element is the container for an inline formatting context which sets the 
-inline wrapping width, and the `<span id="meta_object">` gets any meta object styling (in this
-case, nothing), and contains the formatting for the text runs. The text run styles on the word
-"brown" are applied to its immediate containing span. More precisely: since each text object
-in the array has the potential to be styled, the following markup better represents 
-the semantic equivalent:
-
 ```html
 <div style="width:150px">
```

Comment:
Also add `height:100px` to match your updated JS example.

Reply:
Good catch. Added.